src/_common/content/content-viewer/content-viewer.ts

```diff
@@ -79,6 +94,43 @@ export default class AppContentViewer extends Vue implements ContentOwner {
 	setContent(content: ContentDocument) {
 		this.data = content;
 		this.hydrator = new ContentHydrator(content.hydration);
+		const processedItems: any[] = [];
+
+		this.data.content.forEach(item => {
+			if (item.type === 'mediaItem' && !item.attrs.href) {
+				processedItems.push(
+					new AppContentMediaItem({
```

Comment:
Instead of this utter insanity, we should export the creation of the media item object from the `created` function of that vue component and just call that function here and in the create function.